src/olympia/core/logger.py

```diff
@@ -31,25 +31,32 @@ def process(self, msg, kwargs):
         return msg, kwargs
 
 
-class Formatter(logging.Formatter):
-    """Formatter that makes sure REMOTE_ADDR and USERNAME are available.
-
-    Relies on AmoLoggerAdapter to make sure those variables will be set."""
-
-    def format(self, record):
-        for name in 'REMOTE_ADDR', 'USERNAME':
-            record.__dict__.setdefault(name, '')
-        return super(Formatter, self).format(record)
-
-
 class JsonFormatter(dockerflow.logging.JsonLogFormatter):
     """Like JsonLogFormatter, but with uid and remoteAddressChain set from
-    current user and ip, following mozlog format.
-
-    See Formatter above for the legacy, console version of this."""
-
-    def format(self, record):
+    current user and ip, following mozlog format, as well as an additional
+    severity field at the root of the output for stackdriver."""
+
+    # Map from Python logging levels to Stackdriver severity levels
+    STACKDRIVER_LEVEL_MAP = {
+        # 800 is EMERGENCY but Python doesn't have that
+        # 700 is ALERT but Python doesn't have that
+        logging.CRITICAL: 600,
+        logging.ERROR: 500,
+        logging.WARNING: 400,
+        # 300 is NOTICE but Python doesn't have that
+        logging.INFO: 200,
+        logging.DEBUG: 100,
+        logging.NOTSET: 0,
```

Comment:
I wonder if the stackdriver values should be constants as well but I think for the little amount of usage (here + the tests) it should be fine.

Reply:
They are technically defined in `google-cloud-python` lib but I didn't want to depend on it just for those constants... And yeah, since it's self contained and only used to match `logging` constants I figured redefining them wasn't necessary.